### DISCUSSION TITLE
SDK: Option ** not supported in thread safe mode

### DISCUSSION THREAD

**Discussion by anykno** [projectdiscovery/nuclei#5645] [Q&A]
code snippet

```go
        ......
	eng, err := nuclei.NewThreadSafeNucleiEngineCtx(
		context.Background(),
		nuclei.DisableUpdateCheck(),
		nuclei.WithInteractshOptions(nuclei.InteractshOpts{
			NoInteractsh: true,
		}),
	)
	if err != nil {
		log.Fatal(err)
	}
        ......

```

Result:

```
2024/09/19 00:32:05 Option WithInteractshOptions not supported in thread safe mode
```

There's some options not supported in `ThreadSafe` mode, as I understand, this restriction is to prevent modification on engine internal state when these options provided in `ExecuteNucleiWithOptsCtx` which may be called by multiple thread/routine. But same option in `NewThreadSafeNucleiEngineCtx` do not cause thread safety issues. Is it possible to ignore thread safe restriction while NewThreadSafeNucleiEngine*?, or I missed other important. tks

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
